dynamic/dynamic_message_test.go

```diff
@@ -1952,6 +1953,48 @@ func TestMergeFrom(t *testing.T) {
 	// TODO
 }
 
+func TestBuiltMessage(t *testing.T) {
+	md, err := builder.NewMessage("Person").
+		AddField(builder.NewField("name", builder.FieldTypeString())).
+		Build()
+
+	if err != nil {
+		t.Fatalf("error building Person: %v", err)
+	}
+
+	msg := NewMessage(md)
+	msg.SetFieldByNumber(1, "Cailin")
+
+	_, err = desc.LoadMessageDescriptorForMessage(msg)
+	if err != nil {
+		t.Fatalf("error loading Person's message descriptor: %v", err)
+	}
+}
+
+func TestBuiltNestedMessage(t *testing.T) {
+	submb := builder.NewMessage("Limb").
+		AddField(builder.NewField("length", builder.FieldTypeUInt32()))
+
+	md, err := builder.NewMessage("Person").
+		AddField(builder.NewField("name", builder.FieldTypeString())).
+		AddNestedMessage(submb).
+		Build()
+
+	if err != nil {
+		t.Fatalf("error building Person: %v", err)
+	}
+
+	nmt := md.GetNestedMessageTypes()[0]
+
+	msg := NewMessage(nmt)
+	msg.SetFieldByNumber(1, uint32(40))
```

Comment:
Same remarks as above. Can remove this line and then do stronger verification below instead of just checking there is no error.